libs/prelude/Prelude/Types.idr

```diff
@@ -85,13 +85,19 @@ natToInteger (S k) = 1 + natToInteger k
 -- PAIRS --
 -----------
 
+%inline
+public export
+Bifunctor Pair where
+  bimap f g (x, y) = (f x, g y)
+
+%inline
 public export
 Functor (Pair a) where
-  map f (x, y) = (x, f y)
+  map = mapSnd
 
 public export
-mapFst : (a -> c) -> (a, b) -> (c, b)
-mapFst f (x, y) = (f x, y)
+mapHom : (a -> b) -> (a, a) -> (b, b)
```

Comment:
I am not convinced by this function. Why not have it for all `Bifunctor` or not at all?

Reply:
I don't know, overlooked it